appengine.go

```diff
@@ -10,13 +10,17 @@ package appengine // import "google.golang.org/appengine"
 
 import (
 	"net/http"
+	"path/filepath"
+	"runtime"
 
 	"github.com/golang/protobuf/proto"
 	"golang.org/x/net/context"
 
 	"google.golang.org/appengine/internal"
 )
 
+var MainPath string
```

Comment:
IIUC, it is not necessary to add an exported symbol here. This state could be stored in an internal package instead. Regardless, a doc comment is in order. It's especially worth noting that this might not be set.